content/training/R_Basics/Advanced_Functions.Rmd

```diff
@@ -196,10 +196,150 @@ This is often used to inspect which rows are unique to a data frame. An anti joi
 ![](/images/anti-join.gif)
 
 
+## Graphics with the **ggplot2** package  
 
-TODO: ggplot()  
 
-TODO: for loops  
+R has the capability to make beautiful and versatile graphics using the package **ggplot2** and its function `ggplot()`. Below is a brief overview of the use of `ggplot()`. **There are too many aspects of `ggplot()` to cover here, so it is highly encouraged that you download and review [this ggplot cheatsheet](/public_data/ggplot2-cheatsheet.pdf). **  
 
+For further reading, see [this data visualization tutorial](https://r4ds.had.co.nz/data-visualisation.html) and [this gallery of visualizations and extensions](http://www.ggplot2-exts.org/gallery/) for inspiration.  
 
+### Core components  
 
+A `ggplot()` command is typically built using sub-commands linked by `+` symbols. The sub-commands are executed one-after-another to build the final plot. The core components to provide are:  
+
+* **a data set** - can be a standard data frame, or a spatial shapefile  
+* **coordinate system** - the x and y axes reflect the range of values within specified variables
+* **geom** - the geometric marks to visualize the data (e.g. points, lines, polygons, etc)  
+* **aesthetics** - linking attributes such as color and size of geoms to variables in the data set  
+
+The first command `ggplot()` creates an empty coordinate system to which layers of visualization and nuance are added.  
+* Its first argument specifies the data set to use.  
+* Its `aes()` argument establishes the variables to be mapped to the x and y axes.  
+
+Secondly, the geom is specified, indicating whether the data should be displayed as points, lines, bars, a map, etc. Without a geom, the data will not be visualized and the coordinate plane will be empty.  
+
+Other optional sub-commands can address the scale, labels, aesthetic themes and colors, etc. Note there are other ways to structure a ggplot() command, but the way described above is used in the template. When making a graphic, it often helps to look at examples online and mimic the code that produced them.  
```

Comment:
I think the second two sentences are a bit of distraction. If they browse the ggplot2 gallery, they will see that there are different ways of constructing these. 

```suggestion
Other optional sub-commands can address the scales (e.g `scale_fill_brewer()` chooses a scale for the `fill` aesthetic based on the [Color Brewer 2 color scales](http://colorbrewer2.org/)), labels (e.g. `labs(title = "Attack Rate by Region")` to define a specific title for the plot), themes (e.g. `theme(text = element_text(size = 18))` makes all the text in the plot be a minimum size of 18 points), etc. 
```